src/requests/GroupsApi.ts

```diff
@@ -0,0 +1,1246 @@
+        groupId,
+        title, // Min 1 chars
+        text, // Min 1 chars
+        imageId,
+        sendNotification,
+    }: VRCAPI.Groups.Requests.CreateGroupAnnouncementRequest): Promise<VRCAPI.Groups.Models.GroupAnnouncement> {
+
+        // check parameters
+        if (title.length < 1) return Promise.reject(new Error('Title must be at least 1 character!'));
+        if (text.length < 1) return Promise.reject(new Error('Text must be at least 1 character!'));
+        
+        const body: VRCAPI.Generics.dataKeysCreateGroupAnnouncement = { title, text }
+
+        if (imageId) body.imageId = imageId;
+        if (sendNotification) {
+            body.sendNotification = sendNotification;
+        } else {
+            body.sendNotification = true;
+        }
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.createGroupAnnouncement,
+            pathFormated: ApiPaths.groups.createGroupAnnouncement.path.replace('{groupId}', groupId),
+            body: body,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.GroupAnnouncement>(paramRequest);
+
+        return queryResult;
+
+    }
+
+    /**
+     * Deletes the announcement for a Group.
+     * @param param0 The parameters for the request.
+     * @returns A RequestSuccess object.
+     */
+    public async deleteGroupAnnouncement({groupId}:VRCAPI.Groups.Requests.deleteGroupAnnouncementRequest): Promise<VRCAPI.Generics.RequestSuccess> {
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.deleteGroupAnnouncement,
+            pathFormated: ApiPaths.groups.deleteGroupAnnouncement.path.replace('{groupId}', groupId),
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Generics.RequestSuccess>(paramRequest);
+
+        return queryResult;
+
+    }
+
+    /**
+     * Returns a list of audit logs for a Group.
+     * @param param request parameters for the request
+     */
+    public async getGroupAuditLogs({
+        groupId, 
+        n,  // 1 to 100
+        offset, // 0 to infinity
+        startDate, 
+        endDate,    
+    }: VRCAPI.Groups.Requests.getGroupAuditLogsRequest): Promise<VRCAPI.Groups.Models.GroupAudit> {
+        const parameters: URLSearchParams = new URLSearchParams();
+
+        if (n) {
+            if (n >= 1 && n <= 100) {
+                parameters.append('n', n.toString());
+            } else {
+                parameters.append('n', '60');
+            }
+        } else {
+            parameters.append('n', '60');
+        }
+
+        if (offset) {
+            if (offset >= 0) {
+                parameters.append('offset', offset.toString());
+            } else {
+                console.warn('Offset must be greater than 0! Setting offset to 0.');
```

Comment:
This should throw.

Reply:
not exactly because this isn't a code breaking error, it's just bad data, we can handle it if it's out of range, when it's not valid, it will just use the default value. You think it's still better to throw an error despite this?